clang/lib/CodeGen/CodeGenAction.cpp

```diff
@@ -418,6 +418,8 @@ void BackendConsumer::anchor() { }
 } // namespace clang
 
 bool ClangDiagnosticHandler::handleDiagnostics(const DiagnosticInfo &DI) {
+  if (DI.getSeverity() == DS_Error)
+    HasErrors = true;
```

Comment:
I'm not sure I like the way the "HasErrors" bit works... can we refactor so every diagnostic handler doesn't need its own code to set the HasErrors bits?  The way this is written, it looks like every LLVM-based compiler needs some code for this, not just clang.

Reply:
I created https://github.com/llvm/llvm-project/pull/75889 to improve this case a bit.